classwork/сlasswork2.1/main.cpp

```diff
@@ -0,0 +1,108 @@
+#include <iostream>
+#include <unordered_map>
+#include <fstream>
+#include <string>
+#include <print>
+#include <format>
+
+namespace {
+    constexpr const char* DEFAULT_FILE_NAME = "user_stats.txt";
+    constexpr const char* RESET_COMMAND = "bread";
+    constexpr const char* DELETE_COMMAND = "delete";
+}
+
+class UserStatistics {
+public:
+    explicit UserStatistics(const std::string& statisticsFilePath = DEFAULT_FILE_NAME)
+            : fileName(statisticsFilePath) {
+        loadFromFile();
+    }
+
+    ~UserStatistics() {
+        saveToFile();
+    }
+
+    bool incrementCounterForUser(const std::string& name, std::string& message) {
+        if (name == RESET_COMMAND) {
+            userStats.clear();
+            saveToFile();
+            message = "All history has been exterminated.";
+            return true;
+        }
+
+        if (userStats.find(name) == userStats.end()) {
+            userStats[name] = 1;
+            message = std::format("Welcome, {}!", name);
+            return true;
+        } else {
+            int count = ++userStats[name];
+            message = std::format("Hello again({}), {}!", count, name);
+            return true;
+        }
+    }
+
+    bool resetUser(const std::string& name, std::string& message) {
+        auto it = userStats.find(name);
+        if (it != userStats.end()) {
+            userStats.erase(it);
+            message = std::format("Statistics for {} have been reset.", name);
+            return true;
+        } else {
+            message = std::format("No statistics to reset for {}.", name);
```

Comment:
message is output argument which is a bad practice and that is a signal that something wrong here. the root cause of that is stated in the comment above - that message is a part of business logic and data interpretation which should not be a responsibility of this class. Let's just use return value on higher level to do some decisions on message printing